src/app/pages/RewardPage/hooks/useGetFeesEarnedEvents.ts

```diff
@@ -0,0 +1,65 @@
+import { useMemo } from 'react';
+import { Asset } from 'types';
+import { assetByTokenAddress } from 'utils/blockchain/contract-helpers';
+import { useSelector } from 'react-redux';
+import { selectWalletProvider } from 'app/containers/WalletProvider/selectors';
+import { fixNumber } from 'utils/helpers';
+import { bignumber } from 'mathjs';
+import { AssetsDictionary } from 'utils/dictionaries/assets-dictionary';
+import { useGetContractPastEvents } from 'app/hooks/useGetContractPastEvents';
+
+export const useGetFeesEarnedEvents = () => {
+  const { assetRates, assetRatesLoading, assetRatesLoaded } = useSelector(
+    selectWalletProvider,
+  );
+
+  const { events, loading } = useGetContractPastEvents(
+    'feeSharingProxy',
+    'UserFeeWithdrawn',
+  );
+
+  const feesEarnedEvents = useMemo(
+    () =>
+      events.map(({ returnValues }) => {
+        const asset = getFeesEarnedAsset(returnValues.token);
+        const assetDetails = AssetsDictionary.get(asset);
+        const item = assetRates.find(
+          item => item.source === asset && item.target === Asset.RBTC,
+        );
+        const rate = item ? fixNumber(item.value.rate) : '0';
+
+        const rbtcValue =
+          asset === Asset.RBTC
+            ? bignumber(returnValues.amount).toFixed(0)
+            : bignumber(returnValues.amount)
+                .mul(rate)
+                .div(10 ** assetDetails.decimals)
+                .toFixed(0);
```

Comment:
The same calculation as in `useGetFeesEarnedClaimAmount.ts`. Extract it to a function (e.g. `calculateAssetRbtcValue`) that will accept 2 parameters - `AssetDetails` and `amount`.